File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/shop/repository/InMemoryRepository.java
package com.fixture.shop.repository;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import com.fixture.shop.models.Order;
import com.fixture.shop.models.OrderStatus;
import com.fixture.shop.models.Product;

/**
 * In-memory implementation of Repository.
 */
public class InMemoryRepository implements Repository {

    private final Map<String, Order> orders = new HashMap<>();
    private final Map<String, Product> products = new HashMap<>();

    @Override
    public void saveOrder(Order order) {
        orders.put(order.getOrderId(), order);
    }

    @Override
    public Order getOrder(String orderId) {
        return orders.get(orderId);
    }

    @Override
    public void updateOrderStatus(String orderId, OrderStatus status) {
        Order order = orders.get(orderId);
        if (order != null) {
            order.setStatus(status);
        }
    }

    @Override
    public Product getProduct(String sku) {
        return products.get(sku);
    }

    @Override
    public List<Order> listOrdersByUser(String userId) {
        List<Order> result = new ArrayList<>();
        for (Order order : orders.values()) {
            if (order.getUserId().equals(userId)) {
                result.add(order);
            }
        }
        return result;
    }
}

File: skill_sdk/tests/fixtures/ts-project/src/shop/repository/InMemoryRepository.ts
import { Repository } from './Repository';
import { Order } from '../models/Order';
import { OrderStatus } from '../models/OrderStatus';
import { Product } from '../models/Product';

/**
 * In-memory implementation of Repository.
 */
export class InMemoryRepository implements Repository {
  private readonly orders: Map<string, Order> = new Map();
  private readonly products: Map<string, Product> = new Map();

  saveOrder(order: Order): void {
    this.orders.set(order.getOrderId(), order);
  }

  getOrder(orderId: string): Order | undefined {
    return this.orders.get(orderId);
  }

  updateOrderStatus(orderId: string, status: OrderStatus): void {
    const order = this.orders.get(orderId);
    if (order) {
      order.setStatus(status);
    }
  }

  getProduct(sku: string): Product | undefined {
    return this.products.get(sku);
  }

  listOrdersByUser(userId: string): Order[] {
    const result: Order[] = [];
    for (const order of this.orders.values()) {
      if (order.getUserId() === userId) {
        result.push(order);
      }
    }
    return result;
  }
}
